addons/point_of_sale/models/pos_session.py

```diff
@@ -416,11 +416,7 @@ def _accumulate_amounts(self, data):
 
                 if self.company_id.anglo_saxon_accounting and order.picking_id.id:
                     # Combine stock lines
-                    stock_moves = self.env['stock.move'].search([
-                        ('picking_id', '=', order.picking_id.id),
-                        ('company_id.anglo_saxon_accounting', '=', True),
-                        ('product_id.categ_id.property_valuation', '=', 'real_time')
-                    ])
+                    stock_moves = self.env['stock.picking'].search([('origin', '=', order.name)]).mapped('move_lines').filtered(lambda move: move.company_id.anglo_saxon_accounting and move.product_id.categ_id.property_valuation == "real_time")
```

Comment:
Hi @agr-odoo. If performance is a concern, I guess we should keep the old code, except we change the `('picking_id', '=', order.picking_id.id)` to `('picking_id', 'in', self.env['stock.picking'].search([('origin', '=', order.name)]).ids)`. What do you think?